lib/mp-readline/readline.c

```diff
@@ -473,6 +473,9 @@ void readline_init(vstr_t *line, const char *prompt) {
     #if MICROPY_REPL_AUTO_INDENT
     readline_auto_indent();
     #endif
+// make sure pending exceptions are cleared
+    MP_STATE_VM(mp_pending_exception) = MP_OBJ_NULL;
```

Comment:
I'm not sure I'd do this here. The VM should be handling every pending exception to print the traceback or clear it silently. Is there an exit from somewhere that could clear it instead?